src/Sylius/Component/Payment/Tests/Model/PaymentRequestTest.php

```diff
@@ -0,0 +1,145 @@
+        $this->methodMock = $this->createMock(PaymentMethodInterface::class);
+        $this->paymentRequest = new PaymentRequest($this->paymentMock, $this->methodMock);
+    }
+
+    public function testImplementsSyliusPaymentRequestInterface(): void
+    {
+        $this->assertInstanceOf(PaymentRequestInterface::class, $this->paymentRequest);
+    }
+
+    public function testHasNoHashByDefault(): void
+    {
+        $this->assertNull($this->paymentRequest->getId());
+    }
+
+    public function testHasAPaymentByDefault(): void
+    {
+        $this->assertInstanceOf(PaymentInterface::class, $this->paymentRequest->getPayment());
+    }
+
+    public function testHasAPaymentMethodByDefault(): void
+    {
+        $this->assertInstanceOf(PaymentMethodInterface::class, $this->paymentRequest->getMethod());
+    }
+
+    public function testItsPaymentMethodIsMutable(): void
+    {
+        $this->paymentRequest->setMethod($this->methodMock);
+        $this->assertSame($this->methodMock, $this->paymentRequest->getMethod());
+    }
+
+    public function testItsPaymentIsMutable(): void
+    {
+        $this->paymentRequest->setPayment($this->paymentMock);
+        $this->assertSame($this->paymentMock, $this->paymentRequest->getPayment());
+    }
+
+    public function testHasNewStateByDefault(): void
+    {
+        $this->assertSame(PaymentRequestInterface::STATE_NEW, $this->paymentRequest->getState());
+    }
+
+    public function testItsStateIsMutable(): void
+    {
+        $this->paymentRequest->setState('test_state');
+        $this->assertSame('test_state', $this->paymentRequest->getState());
+    }
+
+    public function testHasCaptureActionByDefault(): void
+    {
+        $this->assertSame(PaymentRequestInterface::ACTION_CAPTURE, $this->paymentRequest->getAction());
+    }
+
+    public function testItsActionIsMutable(): void
+    {
+        $this->paymentRequest->setAction('test_action');
+        $this->assertSame('test_action', $this->paymentRequest->getAction());
+    }
+
+    public function testHasNullPayloadByDefault(): void
+    {
+        $this->assertNull($this->paymentRequest->getPayload());
+    }
+
+    public function testItsPayloadIsMutable(): void
+    {
+        $stdClass = new stdClass();
+        $this->paymentRequest->setPayload($stdClass);
+        $this->assertSame($stdClass, $this->paymentRequest->getPayload());
+    }
+
+    public function testHasEmptyArrayResponseDataByDefault(): void
+    {
+        $this->assertSame([], $this->paymentRequest->getResponseData());
+    }
+
+    public function testItsResponseDataAreMutable(): void
+    {
+        $this->paymentRequest->setResponseData([
+            'foo' => 'bar',
+            'bar' => 'foo'
+        ]);
+        $this->assertSame(
+            ['foo' => 'bar', 'bar' => 'foo'],
+            $this->paymentRequest->getResponseData()
+        );
+    }
+
+    public function testItsCreationDateIsMutable(): void
+    {
+        $date = new DateTime('last year');
```

Comment:
```suggestion
        $date = new \DateTime('last year');
```
check also other files